org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/ZigBeeNetworkStateSerializerImpl.java

```diff
@@ -121,12 +123,30 @@ public synchronized void serialize(final ZigBeeNetworkManager networkManager) {
         }
 
         final File file = getNetworkStateFile();
+        final File temp = new File(file.getAbsolutePath() + ".new");
+        final File backup = new File(
+                file.getAbsolutePath() + "." + new SimpleDateFormat("yyyyMMdd.HHmmss.SSS").format(new Date()));
 
         try {
-            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(file), "UTF-8"));
+            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(temp), "UTF-8"));
             stream.marshal(destinations, new PrettyPrintWriter(writer));
             writer.flush();
             writer.close();
+
+            if (file.exists() && !file.renameTo(backup)) {
+                logger.error("Error moving out old network state file (new state will not be saved)");
+                temp.delete();
+            } else if (!temp.renameTo(file)) {
+                logger.error("Error renaming new network state file. Recovering backup (new state will not be saved)");
+                if (backup.exists() && !backup.renameTo(file)) {
+                    logger.error("Error recovering backup network state file");
```

Comment:
Well, arrived at this point, we lost the network state file. It was renamed, and renaming it back to the the original name failed: there is something really wrong with the setup here.
I would keep it as error.

Reply:
I don't really agree - what does the user do? Do you think it fits the criterea?

error logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore

Since the system will normally recover after the loss of the file, I'm not sure this is correct.

Just because we renamed the file a couple of times, doesn't make any difference (IMHO) - the level of logging needs to be proportionate to the result - ie does the system still work or is it now completely hosed?

Reply:
Well, if the files in the the config directory cannot be renamed, there is definetely something really wrong in the installation. And I was also trying to be consistent with the existing code, as if the file write fails due to an exception, it is reported as error. The only difference here is that file open / write / close operations launch an exception in case of error, while renames and deletes return false instead.

But I have anyway changed it to warn, as requested.